### PR TITLE
Add rake task to run payment calculator to get payment breakdown for a lead provider

### PR DESCRIPTION
**Pull request by plisovin**
Depends on [this branch](https://github.com/DFE-Digital/early-careers-framework/tree/CPDTP-131)

```
rake payment_calculation:breakdown Capita
```

Result:

```
{
  "service_fees": {
    "service_fee_per_participant": "316.67",
    "service_fee_total": "633339.0",
    "service_fee_monthly": "21839.28"
  },
  "output_payment": {
    "per_participant": "587.4",
    "start": {
      "retained_participants": 0,
      "per_participant": "117.48",
      "subtotal": "0.0"
    }
  }
}
```

### CODE REVIEW

File: db/seeds/sandbox_data.rb [hunk 1]
```diff
@@ -18,15 +18,16 @@ def generate_provider_token(lead_provider, cohort, logger)
...
       user = User.create!(full_name: Faker::Name.name, email: Faker::Internet.email)
 
-      school = School.create!(
-        urn: Faker::Internet.uuid,
-        name: Faker::Company.name,
-        address_line1: Faker::Address.street_address,
-        postcode: Faker::Address.postcode,
-      )
+      school = School.find_or_create_by!(
+        urn: sprintf("%06d", (10_000 + i)),
```
Comment (kicferk1) [inline db/seeds/sandbox_data.rb:25]: What does `sprintf` do here? Make sure we have 6 digits in a string?

File: lib/tasks/payment_breakdown.rake [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+namespace :payment_calculation do
+  desc "run payment calculator for a given lead provider"
+  task breakdown: :environment do
+    logger = Logger.new($stdout)
+    logger.formatter = proc do |_severity, _datetime, _progname, msg|
+      "#{msg}\n"
+    end
+
+    begin
+      lead_provider = LeadProvider.find(ARGV[1])
```
Comment (kicferk1) [inline lib/tasks/payment_breakdown.rake:12]: Instead of using `ARGV`, could we do 
```
task breakdown:, [:lead_provider_name] => :environment do |_task, args|
  lead_provider_name = args.lead_provider_name
```
or something similar?

File: lib/tasks/payment_breakdown.rake [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+namespace :payment_calculation do
+  desc "run payment calculator for a given lead provider"
+  task breakdown: :environment do
+    logger = Logger.new($stdout)
+    logger.formatter = proc do |_severity, _datetime, _progname, msg|
+      "#{msg}\n"
+    end
```
Comment (kicferk1) [inline lib/tasks/payment_breakdown.rake:9]: The more I see loggers and puts all over the place the more I wonder if we should have a common way of doing them. Oh well, something for later.

File: lib/tasks/payment_breakdown.rake [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    rescue StandardError
+      lead_provider = LeadProvider.find_by(name: ARGV[1])
+    end
+
+    breakdown = CalculationOrchestrator.call({ lead_provider: lead_provider }, event_type: :start)
+    logger.info JSON.pretty_generate(breakdown)
+  rescue StandardError
+    logger.info "Lead provider for '#{ARGV[1]}' not found"
+  ensure
+    exit(0)
```
Comment (kicferk1) [inline lib/tasks/payment_breakdown.rake:22]: Do we need this? What happens if we don't have it?

File: db/seeds/sandbox_data.rb (hunk 1 above)
Comment (plisovin) [reply db/seeds/sandbox_data.rb:25]: yep, just adding some trailing zeros

File: lib/tasks/payment_breakdown.rake (hunk 2 above)
Comment (plisovin) [reply lib/tasks/payment_breakdown.rake:12]: I personally prefer the ARGV as otherwise you will need to run it using the following:

```
rake "payment_calculation:breakdown[Capita]"
```

instead of a more intuitive:

```
rake payment_calculation:breakdown Capita
```

File: lib/tasks/payment_breakdown.rake (hunk 4 above)
Comment (plisovin) [reply lib/tasks/payment_breakdown.rake:22]: It will treat the command line parameter as a task name and will print an error although after successful execution

File: lib/tasks/payment_breakdown.rake (hunk 2 above)
Comment (kicferk1) [reply lib/tasks/payment_breakdown.rake:12]: Alright.

In that case, could we add a note to readme, or a comment in this file about running this task? Seeing how (again) some tasks use another parameter format?

Comment (plisovin) [reply lib/tasks/payment_breakdown.rake:12]: Add here 17f91d9b

File: db/seeds/sandbox_data.rb [hunk 5]
```diff
@@ -18,15 +18,16 @@ def generate_provider_token(lead_provider, cohort, logger)
...
+      ) do |s|
+        s.name = Faker::Company.name
+        s.address_line1 = Faker::Address.street_address
+        s.postcode = Faker::Address.postcode
+      end
```
Comment (blackrat) [inline db/seeds/sandbox_data.rb:30]: Can we split this into a separate PR? The seed_data update needs to happen with or without this ticket, and in looking at migrations, we may be missing entries in SchoolCohort to correctly pick up the lead_provider for the update.

**Review by blackrat — CHANGES_REQUESTED** [review]

File: db/seeds/sandbox_data.rb (hunk 5 above)
Comment (plisovin) [reply db/seeds/sandbox_data.rb:30]: Ok, this is just a small fix to school urn to have a correct format

Comment (plisovin) [reply db/seeds/sandbox_data.rb]: @blackrat Rolled back

File: README.md [hunk 6]
```diff
@@ -134,6 +134,12 @@ bundle exec rake 'schools:send_invites[urn1 urn2 ...]'
 bundle exec rake lead_provider:generate_token "name or id"
 ```
 
+## Run payment calculator for a given lead provider to generate the payment breakdown
+
+```bash
+bundle exec rake payment_calculation:breakdown "name or id"
```
Comment (kicferk1) [inline README.md:140]: Probably warrants an update now that we added number of participants

Comment (plisovin) [reply README.md]: Thanks, done in c6e947e

**Review by kicferk1 — APPROVED** [review]
